packages/api/cms-api/src/auth/strategies/auth-proxy-jwt.strategy.ts

```diff
@@ -0,0 +1,47 @@
+import { Injectable } from "@nestjs/common";
+import { PassportStrategy, Type } from "@nestjs/passport";
+import { passportJwtSecret } from "jwks-rsa";
+import { ExtractJwt, Strategy } from "passport-jwt";
+
+import { CurrentUserInterface, CurrentUserLoaderInterface } from "../current-user/current-user";
+
+interface AuthProxyJwtStrategyConfig {
+    jwksUri: string;
+    currentUserLoader?: CurrentUserLoaderInterface;
+}
+
+class CurrentUserLoader implements CurrentUserLoaderInterface {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    async load(data: any): Promise<CurrentUserInterface> {
+        return {
+            id: data.sub,
+            name: data.name,
+            email: data.email,
+            language: data.language,
+            role: data.ext?.role,
+            rights: data.ext?.rights,
+        };
+    }
+}
+
+export function createAuthProxyJwtStrategy(config: AuthProxyJwtStrategyConfig): Type {
```

Comment:
Why did you decide to go for factories? It would also be possible to use the class with constructor: `new AuthProxyJwtStrategy(config)`

Furthermore, the "Nest way" for factories would be `AuthProxyJwtStrategyFactory.create(config)`. I know, we're not using this convention consistently (because we've only learnt about it at after starting with Comet), but shouldn't we use it for new functionality?

Reply:
Because we have to provide the class and not an instance. Therefore I decided for the factory instead of the constructor using DI which would need  a config provider that makes it a bit more complicated.

Reply:
Do you have a link for a description for the "Nest way" of factories? I couldn't find anything.

Reply:
> Because we have to provide the class and not an instance.

not if you use useFactory/useValue, I'm not sure if that is really better tough

Reply:
> Do you have a link for a description for the "Nest way" of factories? I couldn't find anything.

I didn't find it either. I believe an example factory was in the docs at one time. However, you can find the style in the Nest source code, for example https://github.com/nestjs/graphql/blob/master/packages/graphql/lib/schema-builder/factories/args.factory.ts

Reply:
Now it uses DI where possible and therefore prevents the usage of factories where possible.

Reply:
Do you like this better than using factories? To be clear: I don't using factories, I mistakenly thought they were unnecessary - which they aren't - that's why I raised the question

Reply:
This is the more Nest.JS way and has the advantage of a real DI (event though we don't take advantage on this). On the other hand the  usage in the application gets a bit more complicated. Overall, I would stick to this.